Content.Server/Speech/EntitySystems/HypomaniaAccentSystem.cs

```diff
@@ -0,0 +1,46 @@
+using System.Text.RegularExpressions;
+using Content.Server.Speech.Components;
+using Content.Server.Chat.Systems;
+using Robust.Shared.Random;
+
+namespace Content.Server.Speech.EntitySystems;
+
+public sealed class HypomaniaAccentSystem : EntitySystem
+{
+    [Dependency] private readonly ChatSystem _chat = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<HypomaniaAccentComponent, AccentGetEvent>(OnAccent);
+    }
+
+    private void OnAccent(EntityUid uid, HypomaniaAccentComponent component, AccentGetEvent args)
+    {
+        var message = args.Message;
+
+        // . => !
+        message = message.Replace(".", "!");
+
+        /// <summary>
+        /// If the last character in the message is not a '?', append a '!' at the end.
+        /// This check must be performed before replacing '?', to avoid double exclamation marks in interrogative sentences.
+        /// <summary>
```

Comment:
You should only use xmldoc comments before things like functions or variables. You can just use regular comments here (or a multi-line comment if you would prefer)

```suggestion
        // If the last character in the message is not a '?', append a '!' at the end.
        // This check must be performed before replacing '?', to avoid double exclamation marks in interrogative sentences.
```